tests/test_lib_utils.py

```diff
@@ -849,4 +850,35 @@ def test_33_determine_logged_in_user(self):
                            "username": "Wyatt Earp",
                            "realm": "Wild West"},
                           {"user": "Dave Rudabaugh",
-                           "realm": "Dodge City"})
\ No newline at end of file
+                           "realm": "Dodge City"})
+
+    def test_34_compare_generic_condition(self):
+
+        def mock_attribute(key):
+            attr = {"a": "10",
+                    "b": "100",
+                    "c": "1000"}
+            return attr.get(key)
+
+        self.assertTrue(compare_generic_condition("a<100",
+                                                  mock_attribute,
+                                                  "Error {0!s}"))
+
+        self.assertTrue(compare_generic_condition("b==100",
+                                                  mock_attribute,
+                                                  "Error {0!s}"))
+
+        # Unknown comparator
+        self.assertFalse(compare_generic_condition("b>=100",
```

Comment:
This does not work since it will be split at the '>' and thus tries to evaluate **`100`** `>` **`=100`** 
In this case we won't even get a warning since the `else` block is never reached.
Using **`!=`** works as a test but still leaves us with the missing warning and unexpected behavior in case of `>=`, `<=`, `=>` or `=<` (This will then compare strings like '1000' > '=100').

Reply:
You are right. The test is wrong. Also the documentation states, that the values "==", ">" and "<" can be used.
I will fix the text.

Reply:
So the test accidentially checks the correct result. I will change the comparator to a real unknown comparator.

Hm, you migh also want to compare the string-value of attribute "b" and check, if it smaller than the string "=100", would you? ;-)

Reply:
done in d7ace787